### ISSUE TITLE
Stuck in Zoom & Pan mode

### ISSUE THREAD

**Issue by jojoleung** [excalidraw/excalidraw#3553]
I try to use other tools (Text, line, polygon, arrow, etc) but they don't work. I can only zoom and pan. The only thing that works is refreshing my window for the tool to reset.

**Comment by dwelle** [comment]
Hello. What's your browser, browser version and device (desktop/mobile)?

**Comment by jojoleung** [comment]
Thank you for following up with me! :)

*Browser*
Google Chrome

*Browser Version*
Version 90.0.4430.93 (Official Build) (x86_64)

*Device*
MacBook Pro (13-inch, 2017, Four Thunderbolt 3 Ports)
macOS Big Sur, Version 11.2.3 (20D91)


I've reproduced the bug many times. It seems to have multiple triggers. One
of them has to do with text (more often in the "unlocked" mode). When I
finish creating a text box and try to make another text box (especially if
the text box is small or layered on top of another object), the tool then
switches to zoom/pan mode and it gets stuck until I hit the refresh button
in the browser. I'm experiencing this bug consistently and having to
refresh through the process.

I can report back if I pick up on any other patterns.

Thanks again!




*JOELLE LEUNG*
Product Designer
She/her/hers
+1 818 388 7006
joelleleung.com <https://www.joelleleung.com/>

**Comment by dwelle** [comment]
Thanks! It's not totally clear to me what you're experiencing. When you say zoom/pan mode, does that manifest in some way other than not being able to switch tool? Does the cursor change to the `hand` cursor? Does the UI change in some way?

If you could record a video that would be great, but I understand it's probably not an option as you likely can't reproduce on command :).

**Comment by jojoleung** [comment]
Sure, happy to record it today. It's very reproducible. I should mention
that I'm experiencing this bug when I use my *wacom tablet. *I haven't
experienced the same problems with my MacBook Pro trackpad. I can also try
with my mouse.


*JOELLE LEUNG*
Product Designer
She/her/hers
+1 818 388 7006
joelleleung.com <https://www.joelleleung.com/>

**Comment by jojoleung** [comment]
I reproduced the bug with my Wacom tablet/stylus. It happens around *00:58 *and
*1:54.*
 Screen Recording 2021-05-09 at 15.41.30.mov
<https://drive.google.com/file/d/1QFTgIPn7nFJbpwp7fnF-ANQTX-_omXPf/view?usp=drive_web>
I'm not running into this issue with my trackpad or mouse. So I'll just
stick these since they work. 😊

I'm assuming this is a pretty niche problem. Not sure how many of your
users use a tablet and stylus! 😂 But in case you want the data, I'm using
the Wacom CTL4100 Intuos.

Thank you, David!


*JOELLE LEUNG*
Product Designer
She/her/hers
+1 818 388 7006
joelleleung.com <https://www.joelleleung.com/>

**Comment by steveruizok** [comment]
I can produce with my Wacom but not consistently. It seems to happen while also using a trackpad or making some other pointer inputs.

**Comment by andrewcaulfield** [comment]
Hi guys,

I'm also experiencing this issue on the latest iPad Pro using the pencil.  It also seems to stem from using the text tool.

**Comment by dwelle** [comment]
Can anyone facing the issue check if it's fixed in here https://excalidraw-git-fixwysiwygblur-excalidraw.vercel.app/?

Thanks!

**Comment by jojoleung** [comment]
I'm having the same issue with my wacom tablet. I repro'd immediately. See
the vid:
 https://drive.google.com/file/d/1iZ_26bUQ0tF1stFJopkVbRFhsCgfrQ7f/view?usp=sharing
<https://drive.google.com/file/d/1iZ_26bUQ0tF1stFJopkVbRFhsCgfrQ7f/view?usp=sharing>


*JOELLE LEUNG*
Product Designer
She/her/hers
+1 818 388 7006
joelleleung.com <https://www.joelleleung.com/>

**Comment by dwelle** [comment]
Thanks. Ok, that issue seems unrelated after all. But I have trouble following what's going on there. You seem to be pressing `ctrl-a` (select all) and `ctrl-shift-a` (chrome search tabs) shortcuts, but I don't suppose that's related.

Anyway, I can't reproduce with my Wacom tablet (intuos 3) or touchpad or mouse, so I'll ask some questions in the hope of narrowing it down:

1. does it happen only when working with your wacom tablet? If so, do you press any of the pen buttons, and if so, what do you have them mapped to and in what order do you press them?
2. when it locks you into zoom/pan mode, does the zooming happen just by you moving the pen/cursor around, or do you have to press anything in order for it to zoom?

We don't exactly have a zoom mode/modifier where you zoom just by moving your pen/cursor on the canvas which is why I'm having a hard time figuring out what might be the issue.

**Comment by jojoleung** [comment]
Yes, I believe it’s unrelated to the shortcuts. I

1.Only with Wacom tablet. I don’t use the buttons on my pen.
2. Zoom is happening with my just moving the pen around the tablet. I’m not
pressing zoom at all.

**Comment by fbartho** [comment]
Could we solve this by adding a button you can tap to force terminate a stuck zoom/pan mode?

I had a great session on iPad today, and thought the bug was fixed, but indeed after app-switching/using the Text Tool as mentioned in #3649 the pan/zoom mode gets stuck enabled.

Note: I’m replicating this on an iPad with the Magic Keyboard and the Apple Pencil

**Comment by dwelle** [comment]
The problem is, as mentioned in https://github.com/excalidraw/excalidraw/issues/3553#issuecomment-849904373, we don't really have any pan/zoom mode so it's hard to even identify what's going wrong. I thought it's a text-editor-not-submitting issue (https://github.com/excalidraw/excalidraw/issues/3553#issuecomment-849844400) but it seems it's either a completely different issue, or there are multiple issues at play.

**Comment by fbartho** [comment]
@dwelle what do you call this “mode”?
![highlighted screenshot of the copy telling you](https://user-images.githubusercontent.com/209712/120000988-34f71e80-bf88-11eb-913c-9039edbbd4a8.jpeg)

It feels like it’s waiting for some mouse or touch *end* / “up” / “release” event. Could a button be added here to generate whatever event is missing?

**Comment by fbartho** [comment]
Unless that copy is present all the time, and I just didn’t notice 🤦

**Comment by fbartho** [comment]
FYI *Workaround* refreshing the canvas / tab unsticks it, and lets me resume working.

**Comment by dwelle** [comment]
In the context of this discussion, by mode I mean something that changes the interaction with the canvas. For example, when you hold `spacebar`, it changes your cursor and when you drag the canvas you're able to pan it. This "mode" changes the state of the application and modifies how clicking/dragging works for the duration.

There is no such "mode" for zooming (there's cmd/ctrl+wheel to zoom, but that's a distinct action, not a state/mode) which is why I'm having a hard time figuring out what's going on exactly.

The fact I cannot reproduce on any of my devices makes this that much harder :(.

**Comment by fbartho** [comment]
Not sure if this helps, but my iPad, the Magic Keyboard (w/ trackpad) and the Apple Pencil — none of them have a scroll wheel.

is this something to do with the pinch to zoom gesture’s code?

**Comment by dwelle** [comment]
> is this something to do with the pinch to zoom gesture’s code?

Yes, that may be related.

What I can reproduce (on a phone) are weird bugs related to double-click-to-create-text, such as elements being transported close to [0,0] coords when you start dragging them and the pen somehow triggers a double-click.

**Comment by dwelle** [comment]
Can anyone of those affected check if one of these two deployments, in order, either improve or fix the situation? (though I'm not expecting it'll be fixed entirely):

1. https://excalidraw-am3mnsozu-excalidraw.vercel.app/
2. https://excalidraw-9n21j6yq5-excalidraw.vercel.app/

**Comment by fbartho** [comment]
@dwelle It looks like deployment 2 never allows me to enter Text, so that might be a problem ;)

Deployment 1 looks okay, but I can’t reliably (on-command) get stuck in the production version, and I can’t easily get stuck with option 1 either.

Hope this helps!

**Comment by dwelle** [comment]
> It looks like deployment 2 never allows me to enter Text, so that might be a problem ;)

Yes, (2) completely disables double-tap for editing/creating text. I just wanted to figure out the root cause.

> Deployment 1 looks okay, but I can’t reliably (on-command) get stuck in the production version, and I can’t easily get stuck with option 1 either.

I don't understand what you mean by getting stuck. Do you mean getting stuck in the pan/zoom mode? By reliably, you mean it still happens, but not as much?

Note that (1) disables double-tap-to-create text. We wanted to disable it anyway (on mobile) because it was causing other issues as well. Ideally we should still allow double-tap-to-edit text, which (2) disables as well. But if there's not much difference for you between (1) and (2), that's good because we can go with (1). But too bad it doesn't fix it altogether (if I understood you correctly, that is).

**Comment by kyra-chang** [comment]
I have the same issue with my **wacom intuos art**. it's easier to reproduce with the following order:

1. use text tool to type
2. while still in the typing state, use pen to drag (usually did near the text box)
3. stuck in zoom/pan mode

Feel free to check the recoding for reference, I used the (2) link from [here](url) to do the recording but it's the same on the usual website
https://drive.google.com/file/d/1NUgsp_lX4dtvS5AO3n5CF1-x3a-HW-U_/view?usp=sharing

Hope this helps! really like this tool!

**Comment by dwelle** [comment]
Thanks @kyra-chang. So, sadly neither of the two fixes solve the zooming issue :(. I haven't yet managed to repro it anywhere, except maybe once on a Surface Book, but never since.

**Comment by dwelle** [comment]
Made a new PR that should hopefully fix this. You can try here: https://excalidraw-gr0yupojz-excalidraw.vercel.app/

**Comment by dwelle** [comment]
Should be fixed now. Anyone please report if still facing issues.

**Comment by TheodoreHua** [comment]
This issue still happens for me on Firefox, using the main https://excalidraw.com. Drawing works fine until you try to pan using middle click, then all further actions until a refresh just keeps panning. The vercel version linked in this issue (https://excalidraw-gr0yupojz-excalidraw.vercel.app/) works just fine for me, but the main one doesn't.

**Comment by ramfattah** [comment]
Facing this issue right now using MS Edge. I used huion Inspiroy H1060P tablet when i switched between tools and zooming in & out with my laptop(mac) trackpad. Now whenever i draw, it either zooms in or out no matter which tool i chose.

**Comment by tacf** [comment]
Having the same issue with Huion Inspiron 2 running Obsidion excalidraw plugin. Same behaviour as @InfinytRam

**Comment by maxbol** [comment]
Same issue with Wacom Intuos BT S, using the Obsidian plugin

**Comment by dwelle** [comment]
@tacf @maxbol if you can reproduce on excalidraw.com, please open a new issue. If not, you can open an issue on https://github.com/zsviczian/obsidian-excalidraw-plugin/issues. Thanks!

**Comment by carokann** [comment]
--- POSSIBLE SOLUTION/WORKAROUND: double click mouse on Excalidraw canvas.
(Commenting this closed issue since this page seems to be the top hit on a web search.)

I have been having this issue several times with Huion tablet.  I tried refreshing the page, closing/reopening the tab and nothing worked.
I think I have solved it before by restarting the browser.

However, this time I tried double clicking with the mouse on Excalidraw and the problem was resolved.
I will test this solution again next time the issue occurs, and update here.
